File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        t=s.lower()
        f="".join(c for c in t if c.isalnum())
        if not f:
            return True
        l=0
        r=len(f)-1
        while l<r:
            if f[l]!=f[r]:
                return False
            l+=1
            r-=1
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String s2=s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
        int l=0,r=s2.length()-1;
        while(l<r){
            if(s2.charAt(l)!=s2.charAt(r)){
               return false;
            }
            l++;
            r--;
        }
        return true;
    }
}